Pod/Classes/Categories/UIImage+DecodedImage.m

```diff
@@ -69,22 +83,91 @@ + (UIImage *)pin_decodedImageWithCGImageRef:(CGImageRef)imageRef
     }
     
     CGSize imageSize = CGSizeMake(CGImageGetWidth(imageRef), CGImageGetHeight(imageRef));
-    UIGraphicsBeginImageContextWithOptions(imageSize, opaque, 1.0);
-    CGContextRef ctx = UIGraphicsGetCurrentContext();
+    
+    CGBitmapInfo info = opaque ? (kCGImageAlphaNoneSkipFirst | kCGBitmapByteOrder32Host) : (kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Host);
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    
+    //Use UIGraphicsBeginImageContext parameters from docs: https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKitFunctionReference/#//apple_ref/c/func/UIGraphicsBeginImageContextWithOptions
+    CGContextRef ctx = CGBitmapContextCreate(NULL, imageSize.width, imageSize.height,
+                                             8,
+                                             0,
+                                             colorspace,
+                                             info);
+    
+    CGColorSpaceRelease(colorspace);
+    
```

Comment:
Mostly it's to be able to use `UIImage imageWithCGImage:scale:orientation:` so that we can let the UIImage plumbing handling all the transforms related to the orientation.  

Unfortunately the UIGraphics methods only give us access to an opaque CGContext and a resulting UIImage and there is no way to specify the orientation of a UIImage after it's been created.

If you want to continue using the UIGraphics methods and rotate by hand here it can be done, but I don't think there will be any performance benefits and the number of transforms and case statements dealing with that was making my head hurt.